File: 125. Valid Palindrome/isPalindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        if len(s) == 0:
            return True
        charArr = []
        for i in s:
            if ord(i) >= ord('0') and ord(i) <= ord('9'):
                charArr.append(i)
            if ord(i) >= ord('A') and ord(i) <= ord('Z'):
                charArr.append(chr(ord(i) + 32))
            if ord(i) >= ord('a') and ord(i) <= ord('z'):
                charArr.append(i)
        left = 0
        right = len(charArr) - 1
        while left < right:
            if charArr[left] == charArr[right]:
                left = left + 1
                right = right - 1
            else:
                return False
        return True

File: 125. Valid Palindrome/isPalindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        int n = s.length();
        char[] s1 = new char[n];
        int idx = 0;
        for (int i = 0; i < n; i++) {
            if (s.charAt(i) >= '0' && s.charAt(i) <= '9') {
                s1[idx] = s.charAt(i);
                idx++;
            }
            if (s.charAt(i) >= 'A' && s.charAt(i) <= 'Z') {
                int c = s.charAt(i) - 'A';
                s1[idx] = (char)(c + 97);
                idx++;
            }
            if (s.charAt(i) >= 'a' && s.charAt(i) <= 'z') {
                s1[idx] = s.charAt(i);
                idx++;
            }
        }

        int left = 0;
        int right = idx - 1;
        while (left < right) {
            if (s1[left] != s1[right]) {
                return false;
            }
            left++;
            right--;
        }

        return true;
    }
}